src/nvim/os/pipe_process.c

```diff
@@ -0,0 +1,111 @@
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include <uv.h>
+
+#include "nvim/os/uv_helpers.h"
+#include "nvim/os/job.h"
+#include "nvim/os/job_defs.h"
+#include "nvim/os/job_private.h"
+#include "nvim/os/pipe_process.h"
+#include "nvim/memory.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "os/pipe_process.c.generated.h"
+#endif
+
+typedef struct {
+  // Structures for process spawning/management used by libuv
+  uv_process_t proc;
+  uv_process_options_t proc_opts;
+  uv_stdio_container_t stdio[3];
+  uv_pipe_t proc_stdin, proc_stdout, proc_stderr;
+} UvProcess;
+
+void pipe_process_init(Job *job, char **argv, bool enable_stdin,
+    bool enable_stdout, bool enable_stderr)
```

Comment:
These args could be vertically aligned to help readability. I usually end up doing that if they don't all fit on one line.